src/omv/ports/stm32/modules/py_fir_lepton.c

```diff
@@ -0,0 +1,695 @@
+                           bool mirror, bool flip, bool transpose)
+{
+    int kelvin = fir_lepton_get_temperature();
+    const uint16_t *data = fir_lepton_get_frame();
+    int new_min;
+    int new_max;
+
+    if (auto_range) {
+        new_min = INT_MAX;
+        new_max = INT_MIN;
+
+        for (int i = 0, ii = fir_width * fir_height; i < ii; i++) {
+            int temp = data[i];
+
+            if (!fir_lepton_rad_en) {
+                temp = (temp - 8192) + kelvin;
+            }
+
+            if (temp < new_min) {
+                new_min = temp;
+            }
+
+            if (temp > new_max) {
+                new_max = temp;
+            }
+        }
+    } else {
+        float tmp = min;
+        min = (min < max) ? min : max;
+        max = (max > tmp) ? max : tmp;
+        new_min = fast_roundf((min + 273.15f) * 100.f); // to kelvin
+        new_max = fast_roundf((max + 273.15f) * 100.f); // to kelvin
+    }
+
+    float diff = 255.f / (new_max - new_min);
+    int w_1 = fir_width - 1;
+    int h_1 = fir_height - 1;
+
+    for (int y = 0; y < fir_height; y++) {
+        int y_dst = flip ? (h_1 - y) : y;
+        const uint16_t *raw_row = data + (y * fir_width);
+        uint8_t *row_pointer = ((uint8_t *) img->data) + (y_dst * fir_width);
+        uint8_t *t_row_pointer = ((uint8_t *) img->data) + y_dst;
+
+        for (int x = 0; x < fir_width; x++) {
+            int x_dst = mirror ? (w_1 - x) : x;
+            int raw = raw_row[x];
+
+            if (!fir_lepton_rad_en) {
+                raw = (raw - 8192) + kelvin;
+            }
+
+            if (raw < new_min) {
+                raw = new_min;
+            }
+
+            if (raw > new_max) {
+                raw = new_max;
+            }
+
+            int pixel = fast_roundf((raw - new_min) * diff);
+            pixel = __USAT(pixel, 8);
+
+            if (!transpose) {
+                row_pointer[x_dst] = pixel;
+            } else {
+                t_row_pointer[x_dst * fir_height] = pixel;
+            }
+        }
+    }
+}
+
+void fir_lepton_trigger_ffc()
+{
+    if (LEP_RunSysFFCNormalization(&fir_lepton_handle) != LEP_OK) {
+        nlr_raise(mp_obj_new_exception_msg(&mp_type_OSError, "FFC Error!"));
+    }
+}
+
+void fir_lepton_wait_on_ffc(uint n_args, const mp_obj_t *args)
+{
+    uint32_t start = systick_current_millis();
+    uint32_t delay_ms = (n_args < 1) ? 5000 : mp_obj_get_int(args[0]);
+
+    for (;;) {
+        LEP_SYS_STATUS_E status;
+
+        if (LEP_GetSysFFCStatus(&fir_lepton_handle, &status) != LEP_OK) {
+            nlr_raise(mp_obj_new_exception_msg(&mp_type_OSError, "SYS Error!"));
+        }
+
+        if (status == LEP_SYS_STATUS_READY) {
+            break;
+        }
+
+        if ((systick_current_millis() - start) >= delay_ms) {
+            nlr_raise(mp_obj_new_exception_msg(&mp_type_OSError, "Timeout!"));
+        }
+
+        systick_sleep(1);
+    }
+}
```

Comment:
```suggestion
void fir_lepton_trigger_ffc()
{
    if (LEP_RunSysFFCNormalization(&fir_lepton_handle) != LEP_OK) {
        nlr_raise(mp_obj_new_exception_msg(&mp_type_OSError, "FFC Error!"));
    }
}

void fir_lepton_wait_on_ffc(uint n_args, const mp_obj_t *args)
{
    uint32_t start = systick_current_millis();
    uint32_t delay_ms = (n_args < 1) ? 5000 : mp_obj_get_int(args[0]);

    for (;;) {
        LEP_SYS_STATUS_E status;

        if (LEP_GetSysFFCStatus(&fir_lepton_handle, &status) != LEP_OK) {
            nlr_raise(mp_obj_new_exception_msg(&mp_type_OSError, "SYS Error!"));
        }

        if (status == LEP_SYS_STATUS_READY) {
            break;
        }

        if ((systick_current_millis() - start) >= delay_ms) {
            nlr_raise(mp_obj_new_exception_msg(&mp_type_OSError, "Timeout!"));
        }

        systick_sleep(1);
    }
}
```
merge in 1 function.